### PR TITLE
Add DLC Accept Validations

### PR DESCRIPTION
**Pull request by dillionverma**
## What's in this PR

* Add validations for `dlc_accept`
* Add tests for `dlc_accept`
* Bring `dlc_accept` test coverage to 100%

Work based on the [dlcspecs](https://github.com/discreetlogcontracts/dlcspecs/blob/master/Protocol.md#the-offer_dlc-message).

### CODE REVIEW

File: packages/messaging/lib/messages/DlcAccept.ts [hunk 1]
```diff
@@ -126,6 +127,51 @@ export class DlcAcceptV0 extends DlcAccept implements IDlcMessage {
     };
   }
 
+  /**
+   * Validates correctness of all fields in DlcOffer
+   * https://github.com/discreetlogcontracts/dlcspecs/blob/master/Protocol.md#the-offer_dlc-message
+   * @throws Will throw an error if validation fails
+   */
+  public validate(): void {
```
Comment (matthewjablack) [inline packages/messaging/lib/messages/DlcAccept.ts:135]: Would be great to have some validation for these checks

- if any input in funding_inputs is not a BIP141 (Segregated Witness) input.
  - MUST reject the contract.

**Review by matthewjablack — CHANGES_REQUESTED** [review]
Just one small change and then LGTM

File: packages/messaging/lib/messages/DlcAccept.ts (hunk 1 above)
Comment (dillionverma) [reply packages/messaging/lib/messages/DlcAccept.ts:135]: Hey good point, can you take a look at the updated changes again?

File: packages/messaging/lib/messages/DlcAccept.ts [hunk 2]
```diff
@@ -126,6 +127,56 @@ export class DlcAcceptV0 extends DlcAccept implements IDlcMessage {
...
+
+    if (new Set(inputSerialIds).size !== inputSerialIds.length) {
+      throw new Error('inputSerialIds must be unique');
+    }
+
+    // 6. Ensure funding inputs are segwit
+    this.fundingInputs.forEach((input: FundingInputV0) => {
+      if (!input.prevTx.isSegWit)
+        throw new Error('fundingInputs must be segwit');
+    });
```
Comment (matthewjablack) [inline packages/messaging/lib/messages/DlcAccept.ts:177]: Nice this looks good. In the future might be good to write the validation logic inside of fundingInputs, cause then you can just call it for `DlcAccept` and `DlcSign`

**Review by matthewjablack — APPROVED** [review]
LGTM 🥳

File: packages/messaging/lib/messages/DlcAccept.ts (hunk 2 above)
Comment (dillionverma) [reply packages/messaging/lib/messages/DlcAccept.ts:175]: Good callout! Just updated the fundingInput tests and added validate to it.

**Pull request merged**